src/main/java/io/confluent/connect/jdbc/sink/DbStructure.java

```diff
@@ -110,6 +110,15 @@ boolean amendIfNecessary(
     //   We also don't check if the data types for columns that do line-up are compatible.
 
     final TableDefinition tableDefn = tableDefns.get(connection, tableId);
+    if (tableDefn == null) {
+      throw new SQLException(
+          "Table with id " + tableId +  "does not exist in database."
+              + " Since we've either already auto-created the table or confirmed"
+              + " it existed before this point, it must mean this user does not"
+              + " have the permission to read the column data for this table"
+              + " (assuming the database is ACID compliant)."
```

Comment:
this is unnecessary :)
```suggestion
              + " have the permission to read the column data for this table."
```